src/jit/gentree.cpp

```diff
@@ -13167,6 +13161,51 @@ GenTree* Compiler::gtTryRemoveBoxUpstreamEffects(GenTree* op, BoxRemovalOptions
         return nullptr;
     }
 
+    // Handle case where we are optimizing the box into a local copy
+    if (options == BR_MAKE_LOCAL_COPY)
+    {
+        // Retype the box temp to be a struct
+        GenTree* boxTemp = box->BoxOp();
+        assert(boxTemp->IsLocal());
+        const unsigned boxTempLcl = boxTemp->AsLclVar()->GetLclNum();
+        assert(lvaTable[boxTempLcl].lvType == TYP_REF);
+        CORINFO_CLASS_HANDLE boxClass = lvaTable[boxTempLcl].lvClassHnd;
+        assert(boxClass != nullptr);
+
+        JITDUMP("Retyping box temp V%02u to STRUCT\n", boxTempLcl);
+        lvaTable[boxTempLcl].lvType   = TYP_UNDEF;
+        const bool isUnsafeValueClass = false;
+        lvaSetStruct(boxTempLcl, boxClass, isUnsafeValueClass);
+
+        // Remove the newobj and assigment to box temp
+        JITDUMP("Bashing NEWOBJ [%06u] to NOP\n", dspTreeID(asg));
+        asg->gtBashToNOP();
+
+        // Update the copy from the value to be boxed to the box temp
+        //
+        // Copy destination should be *(boxtemp + 4/8) for simple types
+        // Revise to &boxtemp
+        GenTree* copyDst = copy->gtOp.gtOp1;
+
+        if ((copyDst->gtOper == GT_BLK) || (copyDst->gtOper == GT_IND) || (copyDst->gtOper == GT_OBJ))
+        {
+            GenTree* blkArg = copyDst->gtOp.gtOp1;
+            assert(blkArg->gtOper == GT_ADD);
+
+            // Todo: verify addends
```

Comment:
Looks like you wanted to add verification that you are adding +4/+8 here

Reply:
Indeed, thanks for catching that. Added with the latest commit.